src/app/painter.cljc

```diff
@@ -51,13 +52,16 @@
                :style {:width (str canvas-size "px")
                        :height (str canvas-size "px")}})
    (let [ctx (.getContext dom/node "2d")]
-     (e/for [emoji (e/server (e/watch canvas-state))]
+     (e/for [emoji canvas-state]
        (let [x (* (first emoji) scaling-factor)
              y (* (second emoji) scaling-factor)
              emoji (last emoji)
              font-size (str (int (* 30 scaling-factor)) "px")]
          (set! (.-font ctx) (str font-size " sans-serif"))
-         (.fillText ctx emoji x y))))))
+         (.fillText ctx emoji x y)))
+     (when (and (= "canvas" @render-method)
+             (= 0 (count canvas-state)))
```

Comment:
this works as long as you aren't clicking really fast. If somehow we miss the tick where count = 0 – which may even be common in multiplayer with people drawing simultaneously – this is broken. The real fix is to move to SVG as described in slack DM